repos/system_upgrade/el7toel8/libraries/mounting.py

```diff
@@ -0,0 +1,384 @@
+                    with BindMount('/dev', os.path.join(self.target, 'dev')):
+                        yield
+
+        def close(self):
+            """ Releasing the context and perform unmounting """
+            if self.context:
+                next(self.context)
+            self.context = None
+
+        def make_command(self, cmd):
+            """ Transform the command to be executed in the chrooted environment """
+            return [
+                'chroot', self.target
+            ] + cmd
+
+    class NONE(_Implementation):
+        """ Execute the given commands and perform the given operations on the real system and not isolated. """
+        pass
+
+
+class IsolatedActions(object):
+    """ This class allows to perform actions in a manner as if the given base_dir would be the current root """
+
+    def __init__(self, base_dir, implementation, **kwargs):
+        self.base_dir = base_dir
+        self.type = implementation(base_dir, **kwargs)
+
+    def __enter__(self):
+        self.type.create()
+        return self
+
+    def __exit__(self, exception_type, exception_value, traceback):
+        self.type.close()
+
+    def full_path(self, path):
+        """
+        Transform the path given to be prefixed with the base_dir, to get the real path on the system.
+
+        Example: self.base_dir = '/var/lib/leapp/scratch/userspace'
+                 path = '/etc/yum.repos.d/redhat.repo'
+                 The result would be: /var/lib/leapp/scratch/userspace/etc/yum.repos.d/redhat.repo
+        """
+        return os.path.join(self.base_dir, path.lstrip('/'))
+
+    def open(self, path, *args, **kwargs):
+        """
+        Open the path given as if it would be the real system.
+
+        The only difference between this function and the python builtin open is the fact that the path uses
+        self.full_path to translate the passed path argument. All other arguments are passed through
+        """
+        return open(self.full_path(path), *args, **kwargs)
+
+    def call(self, cmd, *args, **kwargs):
+        """ Running the given command using the leapp.libraries.stdlib.run function in a isolated manner. """
+        return run(self.type.make_command(cmd), *args, **kwargs)
+
+    def remove(self, path):
+        """
+        Removes the given file as it would be on the real system.
+        """
+        os.unlink(self.full_path(path))
+
+    def mkdir(self, path, mode=0o777):
+        """
+        Creates the given path as it would be on the real system.
+        """
+        os.mkdir(self.full_path(path), mode=mode)
+
+    def makedirs(self, path, mode=0o777, exists_ok=True):
+        """
+        Creates the whole path recursively for any missing part.
+        """
+        _makedirs(path=self.full_path(path), mode=mode, exists_ok=exists_ok)
+
+    def copytree_to(self, src, dst):
+        """
+        Recursively copy an entire directory tree rooted at src. The destination directory,
```

Comment:
please, keep structure:
'''
title one line

..body..
'''

Reply:
(everywhere)

Reply:
I will fix those in a follow up to not stall this PR any longer